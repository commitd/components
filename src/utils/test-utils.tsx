/* eslint-disable @typescript-eslint/no-explicit-any */
/**
 * Import setupTests for your unit tests and you can use `renderLight` and `renderDark` to render elements wrapped in a ThemeProvider.
 * To render without a theme provider use `renderPlain`.
 *
 * N.B. This adds a simple custom id generator so ids match between snapshots.
 *
 */
import '@testing-library/jest-dom'
import '@testing-library/jest-dom/extend-expect'
import './domrect-polyfill'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import React from 'react'
import { ThemeProvider } from '..'
import ResizeObserver from 'resize-observer-polyfill'

// Use the polyfill for the ResizeObserver.
// This is used in some components.
global.ResizeObserver = ResizeObserver

const LightTheme: React.FC = ({ children }) => (
  <ThemeProvider choice="light">{children}</ThemeProvider>
)

const DarkTheme: React.FC = ({ children }) => (
  <ThemeProvider choice="dark">{children}</ThemeProvider>
)

export const renderPlain = render

export const renderLight = (
  ui: Readonly<React.ReactElement>,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult => render(ui, { wrapper: LightTheme, ...options })

export const renderDark = (
  ui: Readonly<React.ReactElement>,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult => render(ui, { wrapper: DarkTheme, ...options })

// re-export everything
export * from '@testing-library/react'
export { userEvent }
