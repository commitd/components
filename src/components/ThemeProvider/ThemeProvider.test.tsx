import React from 'react'
import { ThemeProvider } from '.'
import { renderPlain } from 'test-utils'

it('renders light without error', () => {
  const { asFragment } = renderPlain(<ThemeProvider />)
  expect(asFragment()).toMatchSnapshot()
})

it('renders dark without error', () => {
  const { asFragment } = renderPlain(<ThemeProvider choice="dark" />)
  expect(asFragment()).toMatchSnapshot()
})
