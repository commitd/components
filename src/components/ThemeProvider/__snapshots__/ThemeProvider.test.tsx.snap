// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dark without error 1`] = `
<DocumentFragment>
  :root,
.sxkmjvb {
  --sx-colors-background: hsl(0,0%,100%);
}

.dark-theme {
  --sx-colors-background: hsl(0,0%,0%);
}

@global {
  text-size-adjust: 100%;
  text-rendering: optimizeLegibility;
}

html {
  overflow-x: hidden;
}

body {
  margin: 0;
}

* {
  margin: 0;
  padding: 0;
}

svg {
  display: block;
}

:root,
.sxkmjvb {
  --sx-colors-background: hsl(0,0%,100%);
}

<div
    class="dark-theme"
  />
</DocumentFragment>
`;

exports[`renders light without error 1`] = `
<DocumentFragment>
  <div
    class="light-theme"
  />
</DocumentFragment>
`;
